canonical-web-and-design/maas-ui
# fix(ui): remove custom routeState handling

## Done
- Remove custom route handing in `App` on `popState`. This was required to work around a bug (see comment in diff), which no longer appears to be present, but also introduced the bug in question here. This has been removed.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Reproduce steps in bug (visit react route, visit angular route, visit user prefs). Ensure no `,` appears in the url, and that you are routed to preferences correctly.
- Ensure old routing bug no longer present (visit machine list, visit angular view, visit settings, ensure settings renders **not** the machine list).

Note: I've QA'd this with a production build on focal against maas master, to ensure the routing issues we were working around are no longer present, but feel free to confirm as well.

## Fixes

Fixes: #1598 

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
